be/src/exprs/vectorized/cast_expr.cpp

```diff
@@ -123,9 +149,20 @@ static ColumnPtr cast_to_json_fn(ColumnPtr& column) {
                 overflow = true;
             }
         } else {
+            if constexpr (AllowThrowException) {
+                THROW_RUNTIME_ERROR_WITH_TYPE(FromType);
+            }
             CHECK(false) << "not supported type " << FromType;
```

Comment:
CHECK here will make the BE crash, it would be better that keep quiet when AllowThrowException is off, right?

Reply:
Change CHECK to DCHECK